### DISCUSSION TITLE
Process for translation update

### DISCUSSION THREAD

**Discussion by coulahan** [diasurgical/DevilutionX#6979] [General]
Hi all, just a brief question - is there any specific process to follow when I want to contribute (in my case, even out a few things that I noticed in the German translation) or can I simply start by creating a fork?

**Comment by DakkJaniels** [comment]
https://github.com/diasurgical/devilutionX/wiki/Translation-Guide

**Reply by coulahan** [reply]
Thanks, that's exactly what I was looking for (but for some reason rather within repo/discussions than in wiki)!

**Comment by AJenbo** [comment]
Starting a fork and making your changes is fine. I would suggest using Poedit for editing the translation, but it's not a requirement. It's nice if you can make 1 commit for updating the translation file, and another with your translations, but not really required either (but may slow down reviewing).

There is also this bit which might be good to give a look:
https://github.com/diasurgical/devilutionX/blob/master/Translations/glossary.md

**Comment by coulahan** [comment]
One more thing - after reading through the *.po file I understand why the item prefixes read a bit odd: The naming system is not supporting grammatical genders. Following up on https://github.com/diasurgical/devilutionX/discussions/5994 I understand that this has been a topic of discussion before. Is there anything planned for future releases? If not, I'd try to play around with the given options trying to enhance it as far as possible.

Edit: Other prefix issue: Some do require a space before central item name, others don't. But I think a workaround might be to hard-code the spaces into the prefix translation.

**Reply by AJenbo** [reply]
No explicit solution defined yet, but we do hope to work on it at some point.

**Reply by coulahan** [reply]
I've tried to understand the way translated names are generated, but failed - I only have limited programming experience in TS rather than c++. In general, I guess a gender parameter per translated item within the translation file that is read if exist or something similar. Perhaps I get into it over time - for now I'm going to stick to update the few translations that appeared odd to me.

**Reply by AJenbo** [reply]
each language would need there own as the gender is not the same in all languages, also some have more then 2 i think. Also there are other complex rules that we would also like to tackle, like when a word should be joined or not when combined. I think @glebm had some ideas for how we should handle it.
